streamly-coreutils.cabal

```diff
@@ -27,14 +27,97 @@ maintainer:          streamly@composewell.com
 copyright:           2020 Composewell Technologies
 category:            Concurrency, Streaming
 stability:           Experimental
-build-type:          Configure
+build-type:          Simple
 
 extra-source-files:  CHANGELOG.md, README.md
 
+common compile-options
+    default-language: Haskell2010
+    ghc-options:      -Wall
+
+common lib-options
+    import: compile-options
+
+common exe-options
+    import: compile-options
+
+common test-options
+    import: compile-options
+
+
 library
-  -- exposed-modules:
-  -- other-modules:
-  -- other-extensions:
-  -- build-depends:
+  import: lib-options
+  build-depends:
+      streamly
+    , base              >= 4.8   && < 5
+    , path              >= 0.8
+    , monad-control     >= 1.0   && < 2
+    , exceptions        >= 0.8   && < 0.11
+    , directory         >= 1.2.2 && < 1.4
   hs-source-dirs:      src
+  exposed-modules:
+                       Streamly.Coreutils.Common
+                     , Streamly.Coreutils.Uniq
+                     , Streamly.Coreutils
+
   default-language:    Haskell2010
+
+-------------------------------------------------------------------------------
+-- Benchmarks
+-------------------------------------------------------------------------------
+
+benchmark coreutils-bench
+  type: exitcode-stdio-1.0
+  hs-source-dirs: Benchmark/
+  main-is: Coreutils.hs
+  build-depends:
+      streamly-coreutils
+    , streamly
+    , base              >= 4.8   && < 5
+    , path              >= 0.8
+    , gauge             >= 0.2.4 && < 0.3
+    , monad-control     >= 1.0   && < 2
+    , exceptions        >= 0.8   && < 0.11
+    , directory         >= 1.2.2 && < 1.4
+
+
+  default-language: Haskell2010
+
+
+-------------------------------------------------------------------------------
+-- Test suites
+-------------------------------------------------------------------------------
+
+test-suite test
+  import: test-options
+  type: exitcode-stdio-1.0
+  main-is: Coreutils.hs
+  js-sources: jsbits/clock.js
+  hs-source-dirs: test/Streamly
+  build-depends:
+      streamly-coreutils
+    , streamly
+    , base              >= 4.8   && < 5
+    , path              >= 0.8
+    , monad-control     >= 1.0   && < 2
+    , exceptions        >= 0.8   && < 0.11
+    , directory         >= 1.2.2 && < 1.4
+  default-language: Haskell2010
+
+-------------------------------------------------------------------------------
+-- Executables
+-------------------------------------------------------------------------------
+
+executable test-streaming
```

Comment:
What is this?

Reply:
To check if the implementation is streaming, using infinite streams. Should I not push this ?

Reply:
No. We can write a test for that. Let it be for now.

Reply:
:+1: okay letting it be